Automattic/woocommerce-services
# Add HAZMAT checkbox and bind to redux state

## Description
This PR adds HAZMAT section to package selection tab if the selected package is from USPS service.
Fixes https://github.com/Automattic/woocommerce-shipping-issues/issues/261

### screenshots/GIFs

![8hRvQ5.png](https://github.com/Automattic/woocommerce-services/assets/789421/323b30ce-f1cc-4a30-8934-ad36234619c4)

### What to test:

- Verify you see the hazmat section ONLY when selecting a USPS package.
- Verify once a package is purchased, you see the HAZMAT keyword on the label.
- Switching from a USPS package to a none-USPS package shouldn't send the `hazmat` in the payload to the connect-server.
